Src/WitsmlExplorer.Api/Services/CredentialsService.cs

```diff
@@ -86,7 +86,7 @@ private async Task<bool> UserHasRoleForHost(string[] roles, Uri host)
 
             bool systemCredsExists = _witsmlServerCredentials.WitsmlCreds.Any(n => n.Host == host);
             IEnumerable<Server> hostServer = allServers.Where(n => n.Url.ToString() == host.ToString());
-            bool validRole = hostServer.Any(n => roles.Contains(n.Role));
+            bool validRole = hostServer.Any(n => n.Roles.Intersect(roles).Any());
```

Comment:
Would Server.Roles ever be empty to allow anybody to access it? In that case, this line would fail.

Reply:
Yes I think it will be empty in some cases but we should probably not grant access based on missing or empty role assignments. When roles are missing or not configured the default will be no access until given explicitly